x/wasm/keeper/msg_server.go

```diff
@@ -44,7 +32,14 @@ func (m msgServer) StoreCode(goCtx context.Context, msg *types.MsgStoreCode) (*t
 		return nil, errorsmod.Wrap(err, "sender")
 	}
 
-	codeID, checksum, err := m.keeper.Create(ctx, senderAddr, msg.WASMByteCode, msg.InstantiatePermission)
+	var policy AuthorizationPolicy
+	if msg.Sender == m.keeper.GetAuthority() {
+		policy = GovAuthorizationPolicy{}
+	} else {
+		policy = DefaultAuthorizationPolicy{}
+	}
```

Comment:
This is the switch that we discussed but seeing this in front of me now in all the methods, I think it would be nice to move this into some helper function like `func selectAuthorizationPolicy(actor string)AuthorizationPolicy` or with a better name